packages/loader/driver-definitions/src/urlResolver.ts

```diff
@@ -38,6 +38,17 @@ export interface IUrlResolver {
     ): Promise<string>;
 }
 
+// Information that can be returned by a lightweight, seperately exported driver function. Used to preanalyze a URL for
+// driver compatibility and preload information.
+export interface DriverPreCheckInfo {
+    // Only needed as long as long as Loader.request() does not work as intended. When Loader.request() caches and
+    // resolves pathing properly, this can be removed. #4489, #4491
+    componentPath: string;
+
+    containerCodeHint?: string;
+    preconnectDomains?: string[];
```

Comment:
Maybe some comments here on what these should be used for. Super excited to see us moving stuff upward.

Reply:
Good point, more things I should have copied over :D